mri/operators/linear/wavelet.py

```diff
@@ -28,21 +28,31 @@ class WaveletN(OperatorBase):
     """ The 2D and 3D wavelet transform class.
     """
 
-    def __init__(self, wavelet_name, nb_scale=4, verbose=0, dim=2, **kwargs):
+    def __init__(self, wavelet_name, nb_scale=4, verbose=0, dim=2,
+                 n_coils=1, n_jobs=1, backend="threading", **kwargs):
         """ Initialize the 'WaveletN' class.
-
         Parameters
         ----------
         wavelet_name: str
             the wavelet name to be used during the decomposition.
         nb_scales: int, default 4
             the number of scales in the decomposition.
+        n_coils: int, default 1
+            the number of coils for calibrationless reconstruction
```

Comment:
typo calibrationless -> multichannel

Reply:
No, we use it only in calibrationless reconstruction, at least for now. Like self-calibrating reconstruction works on multichannel but, we still have one image only on which we do linear operation.

Reply:
I don't understand, the number of coils is precisely the number of channels, so it means that it is for multichannel reconstruction. Calibrationless is just a way to do multichannel reconstruction.

Reply:
What I agree with is that it is used for some multichannel reconstruction. I dont agree that we use n_coils=num_channels for all multichannel reconstructions (self-calibrating reconstruction uses n_coils=1)
Given current state of codes, we use this parameter ONLY for calibrationless reconstruction and hence I named it so. 
However, I will change it given that we could have other reconstructions going forward

Reply:
I don't understand how self calibrating uses only 1 coil

Reply:
(1/2) * sum(||Ft Sl Psi_t Alpha - yl||^2_2,l)

We have only one optimizing image for self calibrating reconstruction. Psi * Alpha gives one image. We apply Sl for every image and then get Fourier coefficients.

Reply:
so we are always only applying the wavelet transform to a single image?

Reply:
Except in calibrationless which optimizes for image in every channel based on proximity defined by Group LASSO or OWL.

Reply:
I see... the name is a bit confusing then but I can't think of a better one rn. For the sake of ending this, let's leave it at that for now.